File: 2405-optimal-partition-of-string/2405-optimal-partition-of-string.java
class Solution {
    public int partitionString(String s) {
         int idx=0;
        int countPartition=1;
        Set<Character> chrHolders=new HashSet<>();
        while (idx<s.length()){
            if(!chrHolders.contains(s.charAt(idx))){
                chrHolders.add(s.charAt(idx));
            }else {
                countPartition++;
                chrHolders=new HashSet<>();
                chrHolders.add(s.charAt(idx));
            }
            idx++;
        }
        return countPartition;
    }
}

File: 2405-optimal-partition-of-string/2405-optimal-partition-of-string.py
class Solution:
    def partitionString(self, s: str) -> int:
        cnt=1
        inV=set()
        for c in s:
            if c not in inV:
                inV.add(c)
                continue
            inV.clear()
            inV.add(c)
            cnt+=1
        
        return cnt
        